### ISSUE TITLE
Fail to run the command `azd pipeline config`

### ISSUE THREAD

**Issue by jerryshia** [Azure/APICenter-Portal-Starter#16]
**Describe the issue:**

Fali to run the command `azd pipeline config --principal-id $AZURE_CLIENT_ID` , the deploy process fails as below:

![image](https://github.com/Azure/APICenter-Portal-Starter/assets/139106314/16953b0d-5280-4f04-a149-1c35f748f27f)

Repro Steps:

1. Run `git clone https://github.com/Azure/APICenter-Portal-Starter.git`
2. Run `azd auth login`.
3. Run `azd up`.
4. Run `gh auth login`.
5. Run `$AZURE_CLIENT_ID = $(./infra/scripts/Get-AzdVariable.ps1 -Key AZURE_CLIENT_ID)`.
6. Run `azd pipeline config --principal-id $AZURE_CLIENT_ID`.

**Environments:**

- OS: Windows
- Nodejs version: v20.5.0
- azd version: 1.9.3 (commit e1624330dcc7dde440ecc1eda06aac40e68aa0a3)

**Expected behavior:**
Run `azd pipeline config --principal-id $AZURE_CLIENT_ID` succesfully

@justinyoo, @hemarina for notification.

**Comment by justinyoo** [comment]
@jerryshia Is it WSL on Windows? How is it possible to run `.sh` on Windows?

**Comment by justinyoo** [comment]
@jerryshia Did you run `azd pipeline config` against your forked repo? You can't run it against `Azure/APICenter-Portal-Starter` because you don't have permissions to write up that repo.

**Comment by justinyoo** [comment]
@jerryshia I'll update the README doc to explicitly mention to fork the repo to use `azd pipeline config`. Thanks for the picking up!

**Comment by jerryshia** [comment]
@justinyoo I have forked the repo to use `azd pipeline config` but it still doesn't work. I used `git update-index --chmod=+x` to add permissions to the files in `./infra/hooks`, and then the execution of `azd pipeline config` succeeded. Is there a better way to handle this ,or is it possible to embed these commands somewhere to eliminate these manual steps?

**Comment by justinyoo** [comment]
@jerryshia Let me try on my end again.

**Comment by justinyoo** [comment]
@jerryshia I did follow exact the same steps you did:

1. Fork this repo to mine
1. Run `git clone https://github.com/<MY_GITHUB_ALIAS>/APICenter-Portal-Starter.git`
1. Run `azd auth login`
1. Run `azd up`
1. Run `gh auth login`
1. Run `$AZURE_CLIENT_ID = $(./infra/scripts/Get-AzdVariable.ps1 -Key AZURE_CLIENT_ID)`
1. Run `azd pipeline config --principal-id $AZURE_CLIENT_ID`
1. Run workflow_dispatch event directly on my GitHub repo

There's no error happening. I did it on both Windows (PowerShell 7.4.x) and WSL (Bash)

> I used `git update-index --chmod=+x` to add permissions to the files in `./infra/hooks`, and then the execution of `azd pipeline config` succeeded.

Where did you do it? You don't need to do this on Windows. Even on Linux box, it's not necessary because all the `.sh` scripts are already executable.
